packages/apps/esm-implementer-tools-app/src/backend-dependencies/openmrs-backend-dependencies.ts

```diff
@@ -88,10 +88,44 @@ function checkIfModulesAreInstalled(
   };
 }
 
-function fetchInstalledBackendModules() {
-  return openmrsFetch(`${restBaseUrl}/module?v=custom:(uuid,version)`, {
-    method: 'GET',
-  });
+async function fetchInstalledBackendModules(): Promise<Array<BackendModule>> {
+  const collected: Array<BackendModule> = [];
+  let nextUrl: string | null = `${restBaseUrl}/module?v=default`;
+  let safetyCounter = 0;
+  const MAX_PAGES = 50;
+
+  const resolveNext = (url?: string | null) => {
+    if (!url) return null;
+    if (/^https?:\/\//i.test(url)) return url;
+    if (url.startsWith('/')) return url;
+    return `${restBaseUrl}/${url.replace(/^\/?/, '')}`;
+  };
+
+  while (nextUrl && safetyCounter < MAX_PAGES) {
+    try {
+      const { data } = await openmrsFetch(nextUrl, { method: 'GET' });
+      const rawResults: Array<any> = Array.isArray(data?.results) ? data.results : [];
+
+      // Use moduleid/moduleId for name matching against required dependency keys (e.g., 'reporting')
+      const pageResults: Array<BackendModule> = rawResults.map((r) => ({
+        uuid: r.moduleId ?? r.moduleid ?? r.uuid,
```

Comment:
While the API can return `moduleId` (but never `moduleid`), we're always requesting the default view where the property will always be `uuid`

```suggestion
        uuid: r.uuid,
```